### PR TITLE
Enhancement/6573 tweak unavailable notice

### PR DESCRIPTION
**Pull request by sashadoes**
## Summary

Addresses issue:

- #6573 

## PR Author Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 5.2 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [ ] Run the code.
- [ ] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [ ] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

File: assets/js/modules/adsense/components/dashboard/DashboardTopEarningPagesWidgetGA4.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
+/**
+ * AdBlockerWarningWidget component.
+ *
+ * Site Kit by Google, Copyright 2021 Google LLC
```
Comment (eugene-manuilov) [inline assets/js/modules/adsense/components/dashboard/DashboardTopEarningPagesWidgetGA4.js:4]:
```suggestion
 * Site Kit by Google, Copyright 2023 Google LLC
```

File: assets/js/modules/adsense/components/dashboard/DashboardTopEarningPagesWidgetGA4.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+		select( CORE_USER ).isItemDismissed( DISMISSED_KEY )
+	);
+
+	if ( isDismissed ) {
+		return <WidgetNull />;
+	}
+
+	return (
+		<SettingsNotice
+			notice="Top earning pages are not yet available in Google Analytics 4."
```
Comment (eugene-manuilov) [inline assets/js/modules/adsense/components/dashboard/DashboardTopEarningPagesWidgetGA4.js:45]: This message should be translatable.

File: assets/js/modules/adsense/components/dashboard/DashboardTopEarningPagesWidgetGA4.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+			dismiss={ DISMISSED_KEY }
+		>
+			Site Kit will notify you as soon as you can connect AdSense and
+			Analytics again.
```
Comment (eugene-manuilov) [inline assets/js/modules/adsense/components/dashboard/DashboardTopEarningPagesWidgetGA4.js:49]: Same here, should be translatable.

File: assets/js/components/SettingsNotice/SettingsNotice.js [hunk 4]
```diff
@@ -33,12 +34,27 @@ import {
...
+const { useSelect, useDispatch } = Data;
 
 export default function SettingsNotice( props ) {
-	const { children, type, Icon = getIconFromType( type ) } = props;
+	const { children, type, dismiss, Icon = getIconFromType( type ) } = props;
+
+	const { dismissItem } = useDispatch( CORE_USER );
+
+	const isDismissed = useSelect( ( select ) =>
+		select( CORE_USER ).isItemDismissed( DISMISSED_KEY )
```
Comment (eugene-manuilov) [inline assets/js/components/SettingsNotice/SettingsNotice.js:49]: If the `dismiss` property is not empty then it contains the slug that we need to check for dismissal. So, you need to pass the `dismiss` property to this selector instead of the constants key.

File: assets/js/components/SettingsNotice/SettingsNotice.js [hunk 5]
```diff
@@ -33,12 +34,27 @@ import {
...
+	);
 
 	const Layout = children ? SettingsNoticeMultiRow : SettingsNoticeSingleRow;
 
+	if ( isDismissed ) {
+		return null;
+	}
```
Comment (eugene-manuilov) [inline assets/js/components/SettingsNotice/SettingsNotice.js:56]: It should return `null` if the `dismiss` property is not empty and `isDismissed` is true.

File: assets/js/components/SettingsNotice/SettingsNotice.js [hunk 6]
```diff
@@ -57,6 +73,17 @@ export default function SettingsNotice( props ) {
 			<div className="googlesitekit-settings-notice__body">
 				<Layout { ...props } />
 			</div>
+			{ dismiss && (
+				<div className="googlesitekit-settings-notice__button">
+					<Button
+						onClick={ () => {
+							dismissItem( DISMISSED_KEY );
```
Comment (eugene-manuilov) [inline assets/js/components/SettingsNotice/SettingsNotice.js:80]: It should dismiss the `dismiss` property value.

File: assets/js/modules/adsense/components/dashboard/DashboardTopEarningPagesWidgetGA4.stories.js [hunk 7]
```diff
@@ -0,0 +1,62 @@
+/**
+ * AdBlockerWarningWidget Component Stories.
+ *
+ * Site Kit by Google, Copyright 2021 Google LLC
```
Comment (eugene-manuilov) [inline assets/js/modules/adsense/components/dashboard/DashboardTopEarningPagesWidgetGA4.stories.js:4]:
```suggestion
 * Site Kit by Google, Copyright 2023 Google LLC
```

**Review by eugene-manuilov — CHANGES_REQUESTED** [review]
Thanks, @sashadoes. This is a good start, but we still need to make some changes. Please, check my comments.

File: assets/js/modules/adsense/components/dashboard/DashboardTopEarningPagesWidgetGA4.stories.js [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+ */
+import { MODULES_ADSENSE } from '../../datastore/constants';
+import { provideModules } from '../../../../../../tests/js/utils';
+import { withWidgetComponentProps } from '../../../../googlesitekit/widgets/util';
+import WithRegistrySetup from '../../../../../../tests/js/WithRegistrySetup';
+import DashboardTopEarningPagesWidgetGA4 from './DashboardTopEarningPagesWidgetGA4';
+
+const WidgetWithComponentProps = withWidgetComponentProps( 'adBlockerWarning' )(
+	DashboardTopEarningPagesWidgetGA4
+);
```
[the remark this answers is not in the document]
Comment (sashadoes) [reply assets/js/modules/adsense/components/dashboard/DashboardTopEarningPagesWidgetGA4.stories.js:30]: @eugene-manuilov I am not sure really. Worth to remove for now 👍🏻

[the remark this answers is not in the document]
Comment (eugene-manuilov) [reply assets/js/modules/adsense/components/dashboard/DashboardTopEarningPagesWidgetGA4.stories.js:30]: Ok, then, please, remove it.

**Comment by sashadoes** [comment]
Thanks, @eugene-manuilov for the review and feedback. I made an update 🙂

File: assets/js/modules/adsense/components/dashboard/AdBlockerWarningWidget.js [hunk 9]
```diff
@@ -1,7 +1,7 @@
 /**
  * AdBlockerWarningWidget component.
  *
- * Site Kit by Google, Copyright 2021 Google LLC
+ * Site Kit by Google, Copyright 2023 Google LLC
```
Comment (eugene-manuilov) [inline assets/js/modules/adsense/components/dashboard/AdBlockerWarningWidget.js:4]:
```suggestion
 * Site Kit by Google, Copyright 2021 Google LLC
```

File: assets/js/googlesitekit/datastore/user/dismissed-items.js [hunk 10]
```diff
@@ -134,6 +134,9 @@ const baseSelectors = {
 	 * @return {(boolean|undefined)} TRUE if dismissed, otherwise FALSE, `undefined` if not resolved yet.
 	 */
 	isItemDismissed: createRegistrySelector( ( select ) => ( state, slug ) => {
+		if ( slug === '' ) {
+			return undefined;
+		}
```
Comment (eugene-manuilov) [inline assets/js/googlesitekit/datastore/user/dismissed-items.js:139]: This change is not needed. You can just omit calling the selector if the `dismiss` property is empty:

```js
const isDismissed = useSelect( ( select ) =>
	dismiss ? select( CORE_USER ).isItemDismissed( dismiss ) : undefined
);
```

Please, revert this change and the one in the tests file.

File: assets/js/modules/adsense/components/dashboard/DashboardTopEarningPagesWidgetGA4.stories.js [hunk 11]
```diff
@@ -0,0 +1,57 @@
...
+					{
+						active: true,
+						connected: true,
+						slug: 'adsense',
+					},
+				] );
+
+				registry
+					.dispatch( MODULES_ADSENSE )
+					.receiveIsAdBlockerActive( true );
```
Comment (eugene-manuilov) [inline assets/js/modules/adsense/components/dashboard/DashboardTopEarningPagesWidgetGA4.stories.js:47]: Is this needed?

File: assets/js/components/SettingsNotice/SettingsNotice.js [hunk 12]
```diff
@@ -57,6 +77,17 @@ export default function SettingsNotice( props ) {
...
 				<Layout { ...props } />
 			</div>
+			{ dismiss && (
+				<div className="googlesitekit-settings-notice__button">
+					<Button
+						onClick={ () => {
+							dismissItem( dismiss );
+						} }
+					>
+						OK, Got it!
```
Comment (eugene-manuilov) [inline assets/js/components/SettingsNotice/SettingsNotice.js:87]: This should be a translatable string.

**Review by eugene-manuilov — CHANGES_REQUESTED** [review]
Thanks, @sashadoes. A few more comments for you.
